examples/platforms/simulation/radio.c

```diff
@@ -628,7 +651,14 @@ void radioSendMessage(otInstance *aInstance)
             *(++timeIe) = (uint8_t)(time & 0xff);
         }
     }
-#endif // OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT && OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+#endif // OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    if (sCslPeriod > 0)
+    {
+        otMacFrameSetCslIe(&sTransmitFrame, (uint16_t)sCslPeriod, getCslPhase());
```

Comment:
Technically, the CSL phase should be measured at SFD. So I think it would be better to be filled at `otPlatRadioTxStarted` although it does not make any differences in simulation platform.

Reply:
I think this relates to another problem: currently data frames and ACKs use a different way to finally update their CSL IE. In `otPlatRadioTxAckStarted`, CSL IE is updated there while in `otPlatRadioTxStarted`, it's not. I think both data frames and ACKs should use the same way to add CSL IE. I remember that there was a discussion about where should the final CSL IE content be generated, in radio layer or in submac layer? I think we should reach an agreement on this. 

In current implementation, for data frames, `CslSampleTime` and `CslPeriod` would be both synchronized to platform radio layer. And hence when it finally updates CSL IE, the radio layer could use a method `getCslPhase` and calcaulate the value itself. For ACK frames, the CSL IE is updated in `otPlatRadioTxAckStarted`, which calls radio callbacks and finally get the value in `SubMac` layer. I think we should unify this.

However, there's another reason for the difference between `otPlatRadioTxAckStarted`  and `otPlatRadioTxStarted`. In https://github.com/openthread/openthread/blob/master/examples/platforms/nrf528xx/src/radio.c#L397, `otPlatRadioTxStarted` is called in `otPlatRadioTransmit` instead of an interrupt context(`nrf_802154_tx_started`). And the main purpose of `otPlatRadioTxStarted` is to start a timer. Meanwhile, `otPlatRadioTxAckStarted` is called in `nrf_802154_tx_ack_started`. In summary, the two methods have a different semantic(meaning) for now. So I'm also usually confused by this difference. I think we should make them the same.

As far as I'm concerned, I think we can let radio layer to generate CSL IE itself and we let `otPlatRadioTxAckStarted` have the same meaning with `otPlatRadioTxStarted`. That means, `otPlatRadioTxAckStarted` is not meaningful for us anymore because we use auto ack now, unlike the way to send a data frame. And we would use other method like `otMacFrameSetCslIe` to update its CSL both for data frames and ACKs, and the same for security stuff updates.

Thoughts? @brianljt @bukepo

Reply:
After It's decided that we would make `otPlatRadioTxAckStarted` would have the similar meaning with `otPlatRadioTxStarted` and things like update CSL IE and process security won't be done in the method. As Ack are automatic in 1.2, `otPlatRadioTxAckStarted` would do nothing. Instead, CSL IE updating and security process would be done by calling `otMacFrameSetCslIe` and security method defined in radio platform.